clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -2472,6 +2472,33 @@ bool SemaHLSL::CheckBuiltinFunctionCall(unsigned BuiltinID, CallExpr *TheCall) {
       return true;
     break;
   }
+  case Builtin::BI__builtin_hlsl_dot2add: {
```

Comment:
So after your PR i'm going to delete this hlsl builtin and replace it with a directX builtin: https://github.com/llvm/llvm-project/issues/132303

To make it easier for me to port your sema check if we could use  SemaRef.<> function only instead of hlsl helper functions id appreciate it. If you can't don't worry about it i'll figure out how to preserve the behavior.

Reply:
As far as I skimmed through SemaRef does not have a direct public function to do this check. Therefore for now I have wrapped the helper function into a lambda to make it easier for the follow up change.